funsor/jax/ops.py

```diff
@@ -64,7 +64,7 @@ def _any(x, dim):
 
 @ops.astype.register(array, str)
 def _astype(x, dtype):
-    return x.astype(dtype)
+    return x.astype(np.result_type(dtype))
```

Comment:
`result_type` can be used to convert x64 to x32 dtypes. In JAX, using `x.astype("int64")` in x32 mode will raise a warning.